### PR TITLE
[Hamming]: Approaches

### PR DESCRIPTION
**Pull request by meatball133**

### CODE REVIEW

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+This approach starts by checking if the two strands are of equal length by using [`len()`][len].
+If not, a [`ValueError`][value-error] is raised.
+
+After that is checked, a variable `count` is initialized to 0.
+The count variable will be used to keep track of the number of differences between the two strands.
+
+This approach uses the [`zip()`][zip] function to iterate over two strings.
+You can read more about how to solve this exercise with `zip()` in the [zip approach][approach-zip].
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:26]:
```suggestion
You can read more about how to solve this exercise with `zip` in the [zip approach][approach-zip].
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+What differs in this approach is that we use a list comprehension to create a list of booleans.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:28]:
```suggestion
What differs in this approach is that we use a generator expression to create booleans.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+The list comprehension iterates over the tuples returned by `zip()`.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:29]:
```suggestion
The generator expression returns an iterator over the tuples returned by `zip`.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Then under the iteration so are the tuples unpacked into two variables, `nucleotide_a` and `nucleotide_b`.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:30]:
```suggestion
Within the iteration, the tuples are unpacked into two variables, `nucleotide_a` and `nucleotide_b`.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+We then compare the characters `nucleotide_a` and `nucleotide_b`.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:31]:
```suggestion
We can then compare `nucleotide_a` and `nucleotide_b`.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+If they are not equal, we add `True` to the list.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:32]:
```suggestion
If they are **not** equal, `True` is produced.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+If they are equal, we add `False` to the list.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:33]:
```suggestion
If they **are** equal, `False` is produced.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+The list comprehension is then passed to the `sum()` function.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:34]:
```suggestion
The generator expression is then passed to the `sum` function.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+The [`sum()`][sum] function will add up all the booleans in the list.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:36]:
```suggestion
`sum` will then iterate over the generator expression and add up all the booleans.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+Where `True` is treated as 1 and `False` is treated as 0.
+You can read more about this behavior in [Boolean as numbers][booleans].
+Then that sum is returned.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:39]:
```suggestion
Finally the totaled booleans are returned.
```

File: exercises/practice/hamming/.approaches/sum/content.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+This approach is also doable with range but it is a bit more verbose:
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/sum/content.md:41]:
```suggestion
This approach is also doable with `range` but it is a bit more verbose:
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    if len(strand_a) != len(strand_b):
+        raise ValueError("Strands must be of equal length.")
+    count = 0
+    for nucleotide_a, nucleotide_b in zip(strand_a, strand_b):
+        if nucleotide_a != nucleotide_b:
+            count += 1
+    return count
+```
+
+This approach starts by checking if the two strands are of equal length by using [`len()`][len].
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:14]:
```suggestion
This approach starts by checking if the two strands are of equal length by using [`len`][len].
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+If not, a [`ValueError`][value-error] is raised.
+
+After that is checked, a variable `count` is initialized to 0.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:17]:
```suggestion
After that is checked, a `<count>` variable is initialized to 0.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+The count variable will be used to keep track of the number of differences between the two strands.
+
+We use [`zip()`][zip] to iterate over the characters in `strand_a` and `strand_b` simultaneously.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:20]:
```suggestion
We use [`zip`][zip] to iterate over the characters in `strand_a` and `strand_b` simultaneously.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+`zip()` is a built in function.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:21]:
```suggestion
`zip` is a built in function.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+It takes any number of iterables and returns an iterator of tuples.
+Where the i-th tuple contains the i-th element from each of the argument iterables.
+For example, the first tuple will contain the first element from each iterable, the second tuple will contain the second element from each iterable, and so on until the shortest iterable is exhausted.
+
+In python are strings are iterables.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:26]:
```suggestion
In Python, strings are iterable.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+Here is an example of using `zip()` to iterate over two strings:
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:28]:
```suggestion
Here is an example of using `zip` to iterate over two strings:
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+
+```python
+>>> zipped = zip("GGACGG", "AGGACG")
+>>> list(zipped)
+[('G', 'A'), ('G', 'G'), ('A', 'G'), ('C', 'A'), ('G', 'C'), ('G', 'G')]
+```
+
+We then use the `zip()` iterator to iterate over the tuples.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:36]:
```suggestion
We then use the `zip` iterator to iterate over the tuples.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+We unpack the tuple into two variables, `nucleotide_a` and `nucleotide_b`.
+You can read more about unpacking in the concept [concept:python/unpacking-and-multiple-assignment]().
+
+We then compare the characters `nucleotide_a` and `nucleotide_b`.
+If they are not equal, we increment the `count` variable by 1.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:41]:
```suggestion
If they are not equal, we increment the count variable by 1.
```

File: exercises/practice/hamming/.approaches/zip/content.md [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+After the loop is finished, we return the `count` variable.
```
Comment (BethanyG) [inline exercises/practice/hamming/.approaches/zip/content.md:43]:
```suggestion
After the loop is finished, we return the count variable.
```

**Review by BethanyG — APPROVED** [review]
Language changes and nits, but otherwise approved.

File: exercises/practice/hamming/.approaches/introduction.md [hunk 21]
```diff
@@ -0,0 +1,95 @@
...
+            count += 1
+    return count
+```
+
+For more information, check the [zip approach][approach-zip].
+
+## Approach: Using sum
+
+Using the built-in [`sum`][sum] removes the need for a counter variable.
+Removing the counter variable makes the code is more concise.
```
Comment (meatball133) [inline exercises/practice/hamming/.approaches/introduction.md:57]:
```suggestion
Removing the counter variable makes the code more concise.
```

**Pull request merged**